ui/ui_v3/scripts/select_table.js

```diff
@@ -0,0 +1,327 @@
+// Functions for table range selection, getting list of headers and filter values from sheet
+
+/**
+ * Detect the table's range based on user's selected cells
+ *
+ * @return {Object} dataRange - The detected table range's details
+ * @return {string} dateRange.inputSheet - The name of the sheet containing detected table
+ * @return {string} dateRange.rangeString - The A1Notation of the range detected
+ * @return {number} dateRange.headerRow - The header row number of the range detected
+ * @return {string} dateRange.headerRange - The header range of the range detected
+ */
+function autoDetectTable() {
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var range = sheet.getDataRange();
+  var rangeActive = sheet.getActiveRange();
+
+  // Getting start and end row, start and end column of the range preselected by user
+  var rowEnd = rangeActive.getLastRow();
+  var rowStart = rangeActive.getRow();
+  var colEnd = rangeActive.getLastColumn();
+  var colStart = rangeActive.getColumn();
+  var maxRows = range.getLastRow();
+  var maxCols = range.getLastColumn();
+  
+  Logger.log('Range of the selected cells');
+  Logger.log(rowStart,rowEnd,colStart,colEnd,maxRows,maxCols);
+  
+  var dataRange;
+
+  // If range selected by user has more than 2 rows and area > 5
+  // detect the selected range as the table range
+  if(rowEnd-rowStart>=2 && (rowEnd-rowStart+1)*(colEnd-colStart+1)>5) {
+    dataRange = [rowStart,rowEnd,colStart,colEnd];
+  }
+  // Else detect the table containing the selected range
+  else {
+    dataRange =  selectTable(rowStart,rowEnd,colStart,colEnd,range.getValues(),maxRows,maxCols);
```

Comment:
this name could be much clear explaining we are trying to detect the table. May be switch the functions selectTable and autoDetectTable. Thoughts ?

Reply:
I agree